src/routers/posts.ts

```diff
@@ -22,25 +22,1077 @@ import {
+ *       200:
+ *         description: A post object.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 _id:
+ *                   type: string
+ *                   description: The unique identifier of the post.
+ *                 title:
+ *                   type: string
+ *                   description: The title of the post.
+ *                 content:
+ *                   type: string
+ *                   description: The content of the post.
+ *                 createdAt:
+ *                   type: string
+ *                   format: date-time
+ *                   description: The creation timestamp of the post.
+ *                 creator_id:
+ *                   type: object
+ *                   properties:
+ *                     _id:
+ *                       type: string
+ *                       description: The unique identifier of the creator.
+ *                     username:
+ *                       type: string
+ *                       description: The username of the post creator.
+ *       404:
+ *         description: Post not found
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 err:
+ *                   type: string
+ *                   description: Error message
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 err:
+ *                   type: string
+ *                   description: Error message
+ */
 router.get('/getPostByID/:postID', getPostByID);
+
+/**
+ * @swagger
+ * /posts/count:
+ *   get:
+ *     tags:
+ *       - Posts operations
+ *     summary: Count all posts
+ *     description: Returns the total number of posts in the database.
+ *     responses:
+ *       200:
+ *         description: The total number of posts.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 count:
+ *                   type: integer
+ *                   description: The total number of posts in the database.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 msg:
+ *                   type: string
+ *                   description: Error message.
+ *                 err:
+ *                   type: string
+ *                   description: Details of the error.
+ */
 router.get('/count', countAllPosts);
+
+/**
+ * @swagger
+ * /posts/search:
+ *   get:
+ *     tags:
+ *       - Posts operations
+ *     summary: Search for posts with filtering, sorting, and pagination
```

Comment:
summary should contains just -
"Must be connected as admin" OR
"Must be connected as user"